lib/OpenQA/UserAgent.pm

```diff
@@ -30,9 +30,13 @@ sub new {
             for my $i (qw(key secret)) {
                 my $attr = "api$i";
                 next if $self->$attr;
-                my $val = $cfg->val($args{api}, $i);
+                # Fetch all the values in the file.
+                my @values = $cfg->val($args{api}, $i);
+                # Keep the latest one
+                my $val = pop @values;
                 next unless $val;
```

Comment:
There's no need to modify the array just to get the last element. I would also remove the comments that just state the obvious. This could boil down to just:

```suggestion
                my @values = $cfg->val($args{api}, $i);
                next unless my $val = $values[-1];
```